src/server/scripts/Kalimdor/HallsOfOrigination/halls_of_origination.cpp

```diff
@@ -0,0 +1,251 @@
+            {
+                switch (eventId)
+                {
+                    case EVENT_FOLLOW_SUMMONER:
+                        me->GetMotionMaster()->MovePoint(0, target->GetPosition(), true);
+                        events.Repeat(Seconds(1));
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+        
+    private:
+        EventMap events;
+        Unit* target;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetHallsOfOriginationAI<npc_hoo_energy_fluxAI>(creature);
+    }
+};
+
+// 82382 - Energy Flux (trash mob Spatial Flux)
+class spell_hoo_energy_flux_target_selector : public SpellScriptLoader
+{
+public:
+    spell_hoo_energy_flux_target_selector() : SpellScriptLoader("spell_hoo_energy_flux_target_selector") { }
+
+    class spell_hoo_energy_flux_target_selector_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_hoo_energy_flux_target_selector_SpellScript);
+
+        void FilterTargets(std::list<WorldObject*>& targets)
+        {
+            // Remove tank
+            if (InstanceScript* instance = GetCaster()->GetInstanceScript())
+                if (Creature* Isiset = ObjectAccessor::GetCreature(*GetCaster(), instance->GetGuidData(DATA_ISISET)))
+                    if (WorldObject* tank = Isiset->AI()->SelectTarget(SELECT_TARGET_TOPAGGRO))
+                        targets.remove(tank);
+
+            targets.remove_if(Trinity::ObjectTypeIdCheck(TYPEID_PLAYER, false));
+            if (targets.empty())
+                return;
+
+            WorldObject* target = Trinity::Containers::SelectRandomContainerElement(targets);
+            targets.clear();
+            targets.push_back(target);
+        }
+
+        void Register() override
+        {
+            OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_hoo_energy_flux_target_selector_SpellScript::FilterTargets, EFFECT_0, TARGET_UNIT_SRC_AREA_ENEMY);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_hoo_energy_flux_target_selector_SpellScript();
+    }
+};
+
+// 74880 - Arcane Energy
+class spell_hoo_arcane_energy_check : public SpellScriptLoader
+{
+public:
+    spell_hoo_arcane_energy_check() : SpellScriptLoader("spell_hoo_arcane_energy_check") { }
+
+    class spell_hoo_arcane_energy_check_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_hoo_arcane_energy_check_AuraScript);
+
+        bool Validate(SpellInfo const* /*spell*/) override
+        {
+            return ValidateSpellInfo({ SPELL_ARCANE_BURST, SPELL_ARCANE_ENERGY });
+        }
+
+        void AfterProc(AuraEffect const* /*aurEff*/, ProcEventInfo& /*eventInfo*/)
+        {
+            if (GetTarget()->GetPowerPct(POWER_ENERGY) == 100.0f)
+            {
+                GetTarget()->CastSpell((Unit*)nullptr, SPELL_ARCANE_BURST, false);
```

Comment:
This cast is needed (Unit*) nullptr?
idk if already ported to master (is not needed anymore in 3.3.5)
and remove this false :)